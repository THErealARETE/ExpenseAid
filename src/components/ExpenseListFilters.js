import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate,setEndDate } from '../actions/filters';
import {DateRangePicker} from 'react-dates'
//import { start } from 'repl';

export class ExpenseListFilters extends React.Component{
  state = {
    calenderFocused: null
  }


  onDatesChange=({ startDate, endDate }) =>{
    this.props.setStartDate(startDate)
    this.props.setEndDate(endDate)
  }
    onFocusChange=calenderFocused=> 
    {this.setState(()=>({calenderFocused}))} 

    onTextChange=(e) => {
       this.props.setTextFilter(e.target.value);
      }
    onSortChange = (e) => {
      if (e.target.value === 'date') {
        this.props.sortByDate();
      } else if (e.target.value === 'amount') {
        this.props.sortByAmount(); 
      }
    }
render (){
return (
  
  <div>
    <input
      type="text"
      value={this.props.filters.text}
      onChange={this.onTextChange}
    />
    <select
      value={this.props.filters.sortBy}
      onChange={this.onSortChange}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
    <DateRangePicker
    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
    startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
    endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
    onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
    focusedInput={this.state.calenderFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
    onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
    numberOfMonths = {1}
    showClearDates={true}
    isOutsideRange={()=>false}
    />
  </div>
)
}
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

// whay are we not calling date and amount in our mapDispatchToProps
const mapDispatchToProps= (dispatch)=>{
  return {
    setStartDate: (startDate)=>dispatch(setStartDate(startDate)),
    setEndDate: (endDate)=>dispatch(setEndDate(endDate)),
    setTextFilter: (text)=>dispatch(setTextFilter(text)),
    sortByDate:()=>dispatch(sortByDate()),
    sortByAmount:()=>dispatch(sortByAmount())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
