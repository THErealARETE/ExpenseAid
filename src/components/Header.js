import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header className= "header">
    <div className = 'content-container'>
    <div className = 'content__header'>
    <Link className ="header__title"  to="/"
     //activeClassName="is-active"
      exact={true}>
    <h1>Expensify</h1>
    </Link>
    <button onClick>logout</button>
    </div>
   </div>
  </header>
);

export default Header;


//<NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>

// <NavLink to="/help" activeClassName="is-active">Help</NavLink>