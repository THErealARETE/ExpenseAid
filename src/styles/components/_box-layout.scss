// yet to create login page and add class name

.box-layout{
    align-items: centre;
    background: url();
    background-size:cover;
    display: flex;
    height: 100 hv;
    width: 100vw;
    justify-content: centre;
}

// use in login page
.box-layout__box{
    background: fade-out(white, .15 );
    border-radius: 3px;  
    padding: $l-size $m-size;
    text-align: center;
    width: 25 rem;
}

//use in title div on login page
.box-layout__title{
    margin: 0 0 $m-size 0;
    line-height: 1;
     
}

// 
