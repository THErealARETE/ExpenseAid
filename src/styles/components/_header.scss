.header{
    background: $dark_blue;
}
.content__header{
    align-items: centre;
    display: flex;
    justify-content: space-between; //try the others
    padding: $s-size, 0 ;
}

.header__title{
color: white;
text-decoration:none;
h1{
    margin: 0 ;

}
}