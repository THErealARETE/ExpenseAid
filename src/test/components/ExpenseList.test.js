import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expense from '../fixtures/expenses';
Enzyme.configure({
    adapter: new Adapter()
})


test('should render expense list with expenses', ()=>{
    const wrapper = shallow(<ExpenseList expenses = {expense}/>)
    expect (wrapper).toMatchSnapshot()
})

test ('should render expense list with empty message',()=>{
    const wrapper = shallow(<ExpenseList expenses = {[]}/>)
    expect (wrapper).toMatchSnapshot()   
})