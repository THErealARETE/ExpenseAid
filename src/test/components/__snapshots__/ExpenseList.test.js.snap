// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list with empty message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseList
    expenses={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          no expenses 
        </p>,
        <h1>
          Expense List
        </h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "no expenses ",
        },
        "ref": null,
        "rendered": "no expenses ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expense List",
        },
        "ref": null,
        "rendered": "Expense List",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            no expenses 
          </p>,
          <h1>
            Expense List
          </h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "no expenses ",
          },
          "ref": null,
          "rendered": "no expenses ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expense List",
          },
          "ref": null,
          "rendered": "Expense List",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render expense list with expenses 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseList
    expenses={
      Array [
        Object {
          "amount": 200,
          "createdAt": 0,
          "description": "Gem",
          "id": "1",
          "note": "",
        },
        Object {
          "amount": 300,
          "createdAt": -345600000,
          "description": "wezzy",
          "id": "2",
          "note": "everything stem ",
        },
        Object {
          "amount": 400,
          "createdAt": 345600000,
          "description": "arm",
          "id": "3",
          "note": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <ExpenseListItem
            amount={200}
            createdAt={0}
            description="Gem"
            id="1"
            note=""
          />,
          <ExpenseListItem
            amount={300}
            createdAt={-345600000}
            description="wezzy"
            id="2"
            note="everything stem "
          />,
          <ExpenseListItem
            amount={400}
            createdAt={345600000}
            description="arm"
            id="3"
            note=""
          />,
        ],
        <h1>
          Expense List
        </h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "amount": 200,
          "createdAt": 0,
          "description": "Gem",
          "id": "1",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "amount": 300,
          "createdAt": -345600000,
          "description": "wezzy",
          "id": "2",
          "note": "everything stem ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "amount": 400,
          "createdAt": 345600000,
          "description": "arm",
          "id": "3",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expense List",
        },
        "ref": null,
        "rendered": "Expense List",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <ExpenseListItem
              amount={200}
              createdAt={0}
              description="Gem"
              id="1"
              note=""
            />,
            <ExpenseListItem
              amount={300}
              createdAt={-345600000}
              description="wezzy"
              id="2"
              note="everything stem "
            />,
            <ExpenseListItem
              amount={400}
              createdAt={345600000}
              description="arm"
              id="3"
              note=""
            />,
          ],
          <h1>
            Expense List
          </h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "amount": 200,
            "createdAt": 0,
            "description": "Gem",
            "id": "1",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "amount": 300,
            "createdAt": -345600000,
            "description": "wezzy",
            "id": "2",
            "note": "everything stem ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "amount": 400,
            "createdAt": 345600000,
            "description": "arm",
            "id": "3",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expense List",
          },
          "ref": null,
          "rendered": "Expense List",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
